### DISCUSSION TITLE
Troubles running pelagicEggDrift with one depth layer from CROCO data.

### DISCUSSION THREAD

**Discussion by N1coleR** [OpenDrift/opendrift#1335] [Q&A] [answered]
Dear OpenDrift community:

Recently, I've been trying to run the PelagicEggDrift model with CROCO information.
I processed .his data in Matlab (see Matlab_code.txt) to only obtain one depth layer and then make a new NC file with
the information of interest.

The main problem is that my particles don´t move (see EggDrift.txt). I've noticed in previous discussions that others have encountered the same issue, but my file is too big to mersh the croco_grd.nc to my NC file. 

Is there any specific variables in croco_grd.nc that I have to add to my NC file?

Also, it seems that my 2D file has some troubles with the ROMS readers from OpenDrift:

19:37:51 WARNING opendrift.readers.reader_ROMS_native:183: Vtransform not found, using 1
19:37:51 WARNING opendrift.readers.reader_ROMS_native:219: 2D dataset, so deleting u and v from ROMS_variable_mapping
19:37:51 INFO    opendrift.readers.reader_ROMS_native:250: 'gls_cmu0'
19:37:51 INFO    opendrift.readers.reader_ROMS_native:251: Did not find complete set of GLS parameters
19:37:51 WARNING opendrift.readers.basereader.structured:50: No proj string or projection could be derived, using 'fakeproj'. This assumes that the variables are structured and gridded approximately equidistantly on the surface (i.e. in meters). This must be guaranteed by the user. You can get rid of this warning by supplying a valid projection to the reader.
19:37:51 INFO    opendrift.readers.basereader.structured:90: Making interpolator for lon,lat to x,y conversion...

other warnings and errors during the run:

19:38:53 INFO    opendrift.models.basemodel.environment:218: Adding a dynamical landmask with max. priority based on assumed maximum speed of 2.0 m/s. Adding a customised landmask may be faster...
19:38:53 INFO    opendrift.models.basemodel.environment:245: Fallback values will be used for the following variables which have no readers: 
19:38:53 INFO    opendrift.models.basemodel.environment:248: 	x_sea_water_velocity: 0.000000
19:38:53 INFO    opendrift.models.basemodel.environment:248: 	y_sea_water_velocity: 0.000000
19:38:53 INFO    opendrift.models.basemodel.environment:248: 	sea_surface_height: 0.000000
19:38:53 INFO    opendrift.models.basemodel.environment:248: 	sea_surface_wave_significant_height: 0.000000
19:38:53 INFO    opendrift.models.basemodel.environment:248: 	sea_ice_area_fraction: 0.000000
19:38:53 INFO    opendrift.models.basemodel.environment:248: 	x_wind: 0.000000
19:38:53 INFO    opendrift.models.basemodel.environment:248: 	y_wind: 0.000000
19:38:53 INFO    opendrift.models.basemodel.environment:248: 	sea_floor_depth_below_sea_level: 100.000000

19:38:53 ERROR   opendrift.models.basemodel.environment:666: 'Reader' object has no attribute 'hc'

19:38:53 WARNING opendrift.models.basemodel.environment:673: Reader roms native is discarded after failing more times than allowed (1)

Any help would be appreciated.
I also attached a pice of my NC file from Matlab, you can download it from: https://www.dropbox.com/scl/fi/rdehbb26ak9tl67amccil/croco06prueba.nc?rlkey=lkr9i72gx6ut8fsin7i5pt8pv&st=7hn3jaqp&dl=0

[EggDrift.txt](https://github.com/user-attachments/files/16007755/EggDrift.txt)
[Matlab_code.txt](https://github.com/user-attachments/files/16007756/Matlab_code.txt)

**Comment by AndresSepulveda** [comment, marked as answer]
Hi,

You do need these variables in your input file. OpenDrift has no problem efficiently reading large files 

What I do to read CROCO files is

1) Rename the time variable
 ncrename -O -v scrum_time,ocean_time  archivo.nc

2) Rewrite the units atttibute 

octave:> nc=netcdf('archivo.nc','w');
octave:> nc{'time'}.units = 'seconds since 2000-01-01 00:00:00';
octave:> nc{'ocean_time'}.units = 'seconds since 2000-01-01 00:00:00';
octave:> tiempo = nc{'time'}(:);
octave:> tiempo = tiempo-tiempo(1);
octave:> nc{'time'}(:)=tiempo;
octave:> nc{'ocean_time'}(:)=tiempo;
octave:> close(nc)

and 
3) Add Vstretching variable. I get it form a  CLM file
with

ncks -v Vstretching croco_clmP_TNZ.nc Vstretching.nc

and then do

ncks -A Vstretching.nc sml_part_mod.nc

**Comment by N1coleR** [comment]
I will try this. Thank you.
